[Website] Committer expectations and process (#306)
apache/arrow-site: committers.md

    {% endif %}
  {% endfor %}
</tbody></table>

> ### **Becoming a committer**

Reviewer: Tiny comment: is the bold font needed here? (the ``**``) The other titles above don't have this

Author: That is an excellent point 👍 

```suggestion
### Becoming a committer
```

There are many ways to [contribute](https://arrow.apache.org/docs/developers/contributing.html)
to the Apache Arrow project, including issue reports,
documentation, tests, and code. Contributors with sustained, high-quality activity
may be invited to become committers by the PMC
as a recognition of their sustained
contribution to the project. A committer can commit

> changes directly in all Arrow github repositories, and have the significant responsibility

Reviewer 2: ```suggestion
changes directly in all Arrow GitHub repositories, and have the significant responsibility
```

for the entire community.

When considering to invite someone to be a committer, the PMC looks for
contributors who are doing the work and exercising the judgment expected
of a committer already. After all, any contributor can do all of the things a
committer does except for merge a PR. While there is no set list of
requirements, nor a checklist that entitles one to commit privileges,
typical behaviors include:

* Contributions beyond pull requests, such as reviewing other pull requests,
  fixing bugs and documentation, triaging issues, answering community
  questions, improving usability, reducing technical debt, helping
  with CI, verifying releases, debugging in strange environments, etc.

* These contributions to the project should be consistent in quality
  and sustained over time, typically on the order of 6 months or more.

* Assistance growing the size and health of the community via
  constructive, respectful, and consensus driven interactions, as
  described in our [Code of Conduct] and [the Apache Way].

The mechanics of how the process works is [documented here].  If you
feel you should be offered committer privileges, but have not been,

> you can reach out to one of the PMC members or the private at arrow
> dot apache dot org mailing list.

Reviewer 2: I think that we don't need to care about spam for `private@arrow.apache.org` because the e-mail address is public.

```suggestion
you can reach out to one of the PMC members or the `private@arrow.apache.org`
mailing list.
```

    {% endif %}
  {% endfor %}
</tbody></table>

### **Becoming a committer**

There are many ways to [contribute](https://arrow.apache.org/docs/developers/contributing.html)
to the Apache Arrow project, including issue reports,
documentation, tests, and code. Contributors with sustained, high-quality activity
may be invited to become a committer by the PMC.

> Becoming a committer is a recognition of sustained

Reviewer 3: This sounds redundant to the previous sentence.

Author: I tried to reduce the redundancy in fb77e71db04

contributors. Becoming a committer is a recognition of sustained
contribution to the project, and comes with the privilege of
committing changes directly in all Arrow github repositories. Becoming
a committer is also a significant responsibility, and committers are
expected to use their status and access to improve the Arrow project
for the entire community.

> Anyone wishing to become a committer can already do all of the things a

Reviewer 3: I feel like this is missing an intro sentence, maybe something like this?

```suggestion
When considering to invite someone to be a committer, the PMC looks for 
contributors who are doing the work and exercising the judgment expected
of a committer. After all, any contributor can already do all of the things a
```

committer does except for merge a PR. While there is no set list of
requirements, nor a checklist that entitles one to commit privileges,
typical behaviors include:

> * Consistent, significant, and sustained involvement in the Arrow

Reviewer 3: Should "involvement" be expanded on? I think we want to be clear that tossing up a bunch of PRs but not otherwise engaging with others' work isn't sufficient, that the significance of the code contributed is not irrelevant but not the top concern.

Reviewer 3: That's better but I don't know that it's enough. What if in addition to this wording change, we swap the order of the following two bullets. and start the next one like: "Contributions are not limited to making pull requests. They include essential project maintenance work such as reviewing pull requests, fixing bugs ..."

Author: Great idea. Thanks @nealrichardson -- I took it one step more in 07d15371c67

```
* Contributions beyond pull requests, such as reviewing other pull requests,
  fixing bugs and documentation, triaging issues, answering community
  questions, improving usability, reducing technical debt, helping
  with CI, verifying releases, debugging in strange environments, etc.

```

* Contributions beyond pull requests, such as reviewing other pull requests,
  fixing bugs and documentation, triaging issues, answering community
  questions, improving usability, reducing technical debt, helping
  with CI, verifying releases, debugging in strange environments, etc.

> * Consistent, significant, and sustained contributions in the Arrow
>   community, typically on the order of 6 months or more.

Reviewer 3: ```suggestion
* These contributions to the project should be consistent in quality
  and sustained over time, typically on the order of 6 months or more.
```

> * Assistance growing the size and health of the community via constructive and

Reviewer 3: Good opportunity to link to the code of conduct, and maybe there's a way to link to some ASF reference on The Apache Way or something.

Author: Great idea -- added in 004f8b77931
